jdaviz/app.py

```diff
@@ -938,12 +937,16 @@ def remove(stack_items):
                 if len(stack.get('children', [])) > 0:
                     stack['children'] = remove(stack['children'])
 
+            empty_stacks = [s for s in stack_items
+                            if not s['viewers'] and not s.get('children')]
+            for empty_stack in empty_stacks:
+                stack_items.remove(empty_stack)
+
             return stack_items
```

Comment:
Can this all be simplified to a single list comprehension?  Or would that limit readability?